packages/neuron-ui/src/components/PageContainer/hooks.ts

```diff
@@ -1,34 +1,72 @@
-import { useCallback, useEffect, useState } from 'react'
+import { TFunction } from 'i18next'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
 import { importedWalletDialogShown } from 'services/localCache'
-import { isDark, openExternal, updateWallet, setTheme as setThemeAPI } from 'services/remote'
+import { isDark, openExternal, setTheme as setThemeAPI, updateWalletStartBlockNumber } from 'services/remote'
 import { Migrate } from 'services/subjects'
 import { getExplorerUrl, isSuccessResponse } from 'utils'
 
+const waitConfirmTime = 5
+
+const useCountDown = (second: number) => {
+  const [countdown, setCountdown] = useState(0)
+  const countDecreaseTimeoutRef = useRef<ReturnType<typeof setTimeout>>()
+  const resetCountDown = useCallback(() => {
+    clearTimeout(countDecreaseTimeoutRef.current)
+    setCountdown(second)
+    const decrement = () => {
+      countDecreaseTimeoutRef.current = setTimeout(() => {
+        setCountdown(v => (v > 0 ? v - 1 : v))
+        decrement()
```

Comment:
Why is `decrement` called recursively

Reply:
Because I use `setTimeout` to count down. Now I change it to `setInterval` and remove the recursively called.